--- a/configure.in	2015-05-06 15:15:32.000000000 +0200
+++ a/configure.in.patched	2015-05-06 15:35:07.000000000 +0200
@@ -721,11 +721,11 @@
      use_arch="mips"
      ffmpeg_target_os=linux
      use_joystick=no
-     use_gles=yes
-     use_sdl=no
+     use_gles=no
+     use_sdl=yes
      use_x11=no
      use_wayland=no
-     use_static_ffmpeg=no
+     use_static_ffmpeg=yes
      ;;
   *)
      AC_MSG_ERROR(unsupported host ($host))
