--- a/configure.ac	2015-05-24 14:22:23.000000000 +0200
+++ a/configure.ac.patched	2015-05-24 14:53:00.000000000 +0200
@@ -1145,22 +1145,7 @@
   AC_CHECK_LIB([mysqlclient], [main],, AC_MSG_ERROR($missing_library))
 fi
 AC_CHECK_LIB([bluetooth],   [hci_devid],, AC_MSG_RESULT([Could not find suitable version of libbluetooth]))
-
-AC_LANG_PUSH([C++])
-PKG_CHECK_MODULES([TINYXML], [tinyxml >= 2.6.2],
-  [INCLUDES="$INCLUDES $TINYXML_CFLAGS"; LIBS="$LIBS $TINYXML_LIBS"],
-  [AC_RUN_IFELSE(
-   [AC_LANG_SOURCE([[
-#include <stdlib.h>
-#include <tinyxml.h>
-
-int main() {
-  if (TIXML_MAJOR_VERSION < 2) exit(1);
-  if (TIXML_MAJOR_VERSION == 2 && ( TIXML_MINOR_VERSION < 6 || ( TIXML_MINOR_VERSION == 6 && TIXML_PATCH_VERSION < 2 ))) exit(1);
-}
-    ]])],[AC_CHECK_LIB([tinyxml], [main],, AC_MSG_ERROR("tinyxml >= 2.6.2 not found"))], [AC_MSG_ERROR([tinyxml >= 2.6.2 not found])]
-  )])
-AC_LANG_POP([C++])
+AC_CHECK_LIB([tinyxml],     [main],, AC_MSG_ERROR($missing_library))
 
 PKG_CHECK_MODULES([YAJL], [yajl >= 2],
   [INCLUDES="$INCLUDES $YAJL_CFLAGS"; LIBS="$LIBS $YAJL_LIBS"; YAJL_FOUND="true"],
